src/core/server.js

```diff
@@ -686,6 +687,7 @@ Server.prototype._handleResponse = function(message) {
       error_message: 'Remote reported an error.',
       remote: message
     });
+    request.emit('response');
```

Comment:
I was imagining `response` as a catch-all... you could put it outside this success/failure check and rename it "message" if that sounds better